components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -242,16 +275,16 @@ export default class NotificationsTab extends React.PureComponent {
         }
     }
 
-    onCustomChange = () => {
-        this.refs.customcheck.checked = true;
+    onCustomChange = (): void => {
+        this.setState({customcheckChecked: true});
         this.updateCustomMentionKeys();
     }
 
-    createPushNotificationSection = () => {
+    createPushNotificationSection = (): JSX.Element => {
         if (this.props.activeSection === 'push') {
-            const inputs = [];
-            let extraInfo = null;
-            let submit = null;
+            const inputs: JSX.Element[] = [];
+            let extraInfo: JSX.Element = <></>;
+            const submit = this.handleSubmit;
```

Comment:
This looks like a functional change from the previous version. I think it adds a save button to the component which we don't want in some cases. Could you change it back so that it's only passed in at the same times as before?